contracts/src/access/test/ZOwnablePK.test.ts

```diff
@@ -0,0 +1,502 @@
+  const rt_type = new CompactTypeVector(4, new CompactTypeBytes(32));
+  const bCounter = convert_bigint_to_Uint8Array(32, counter);
+  const bDomain = new TextEncoder().encode(DOMAIN);
+
+  const commitment = persistentHash(rt_type, [
+    id,
+    instanceSalt,
+    bCounter,
+    bDomain,
+  ]);
+  return commitment;
+};
+
+const buildCommitment = (
+  pk: ZswapCoinPublicKey,
+  nonce: Uint8Array,
+  instanceSalt: Uint8Array,
+  counter: bigint,
+  domain: string,
+): Uint8Array => {
+  const id = createIdHash(pk, nonce);
+
+  const rt_type = new CompactTypeVector(4, new CompactTypeBytes(32));
+  const bCounter = convert_bigint_to_Uint8Array(32, counter);
+  const bDomain = new TextEncoder().encode(domain);
+
+  const commitment = persistentHash(rt_type, [
+    id,
+    instanceSalt,
+    bCounter,
+    bDomain,
+  ]);
+  return commitment;
+};
+
+describe('ZOwnablePK', () => {
+  describe('before initialize', () => {
+    it('should fail when setting owner commitment as 0', () => {
+      expect(() => {
+        const badId = new Uint8Array(32).fill(0);
+        new ZOwnablePKSimulator(badId, INSTANCE_SALT, isInit);
+      }).toThrow('ZOwnablePK: invalid id');
+    });
+
+    it('should initialize with non-zero commitment', () => {
+      const notZeroPK = utils.encodeToPK('NOT_ZERO');
+      const notZeroNonce = new Uint8Array(32).fill(1);
+      const nonZeroId = createIdHash(notZeroPK, notZeroNonce);
+      ownable = new ZOwnablePKSimulator(nonZeroId, INSTANCE_SALT, isInit);
+
+      const nonZeroCommitment = buildCommitmentFromId(
+        nonZeroId,
+        INSTANCE_SALT,
+        INIT_COUNTER,
+      );
+      expect(ownable.owner()).toEqual(nonZeroCommitment);
+    });
+  });
+
+  describe('when not initialized correctly', () => {
+    const isNotInit = false;
+
+    beforeEach(() => {
+      ownable = new ZOwnablePKSimulator(
+        randomByteArray,
+        INSTANCE_SALT,
+        isNotInit,
+      );
+    });
+    type FailingCircuits = [method: keyof ZOwnablePKSimulator, args: unknown[]];
+    const randomByteArray = new Uint8Array(32).fill(123);
+    const randomCounter = 321n;
+    // Circuit calls should fail before the args are used
+    const circuitsToFail: FailingCircuits[] = [
+      ['owner', []],
+      ['assertOnlyOwner', []],
+      ['transferOwnership', [randomByteArray]],
+      ['renounceOwnership', []],
+      ['_computeOwnerCommitment', [randomByteArray, randomCounter]],
+      ['_transferOwnership', [randomByteArray]],
+    ];
+    it.each(circuitsToFail)('%s should fail', (circuitName, args) => {
+      expect(() => {
+        (ownable[circuitName] as (...args: unknown[]) => unknown)(...args);
+      }).toThrow('Initializable: contract not initialized');
+    });
+
+    it('should allow pure computeOwnerId', () => {
+      const eitherOwner = utils.createEitherTestUser('OWNER');
+
+      expect(() => {
+        ownable._computeOwnerId(eitherOwner, randomByteArray);
+      }).not.toThrow();
+    });
+  });
+
```

Comment:
What do you think about adding a `describe('does not generate ownerId with SHA256')` and validating that all functions that make calls to `assertOnlyOwner` fail? I think these tests would make it clear that the user supplied `ownerId` MUST use SHA256. Otherwise, circuit verification would fail.

Reply:
I think that's a very good test scenario to include. On it!